src/main/java/org/sagebionetworks/web/server/servlet/oauth2/OAuth2SessionServlet.java

```diff
@@ -91,6 +92,16 @@ public void validateUser(
     } catch (SynapseNotFoundException e) {
       // used to send the user to register
       resp.sendRedirect(REGISTER_ACCOUNT);
+    } catch (SynapseTwoFactorAuthRequiredException e) {
+      // Go back to the login page to attempt the 2fa challenge.
+      resp.sendRedirect(
+        LOGIN_PLACE +
+        "0" +
+        "?userId=" +
+        e.getUserId().toString() +
+        "&twoFaToken=" +
+        e.getTwoFaToken()
+      );
```

Comment:
Future changes in synapse-react-client will have the login component check these parameters and use them to jump into 2fa straight from an OAuth login attempt.